presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleTtlNodeSelector.java

```diff
@@ -261,6 +261,7 @@ private Duration getEstimatedExecutionTimeRemaining()
                 .map(remoteTask -> nodeMap.getActiveNodesByNodeId().get(remoteTask.getNodeId()))
                 // nodes may sporadically disappear from the nodeMap if the announcement is delayed
                 .filter(Objects::nonNull)
+                .filter(ttlInfo::containsKey)
```

Comment:
My understanding is without this, it could fail in the line below--ttlInfo.get(node) will return null (default value for map), and calling `isPresent` on it will throw.

Reply:
got it.

Reply:
`ttlInfo` is a map from `InternalNode` to `Optional<ConfidenceBasedTtlInfo>`. This is the opposite of the `nonNull` check above, i.e, nodes in the nodeMap may not have their TTL information fetched yet and so won't be present in the `ttlInfo` map, resulting in the NPE.